content/2024/duckdb-intro.rst

```diff
@@ -0,0 +1,167 @@
+:blogpost: true
+:date: 2024-02-13
+:author: Richard Darst
+:category: software
+
+
+DuckDB: an intro
+================
+
+A colleague recently pointed me to `DuckDB <https://duckdb.org/>`__ as
+an improvement on SQLite for my work.  After many months, I have
+checked more and what I found was really useful.  This blog posts is
+what I wish I had long ago.
+
+DuckDB doesn't replace SQLite.  I look at it as a natural companion
+that works together for another type of workload.
+
+
+
+The basics
+----------
+
+DuckDB is "sqlite but optimized for analytics instead of
+transactions".  It has all the best things of SQLite, and SQLite is
+the right mental model for understanding it:
+
+- Serverless, in-process library
+- Database = single file
+- "Normal" SQL
+- Lightweight code, C++, easily embeddable in anything
+- Seems to value stability and portability
+
+The extra advantages of DuckDB are:
+
+- Column oriented, which makes it much faster for analytics
```

Comment:
what means "column oriented"? or perhaps, column oriented in contrast to what typical other setup?